services/blockstorage/test/get_block_headers.go

```diff
@@ -0,0 +1,105 @@
+package test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+)
+
+var _ = Describe("Block storage", func () {
+	When("asked to provide transactions block header", func () {
+		It("returns transactions block header", func () {
+			driver := NewDriver()
+			driver.expectCommitStateDiff()
+
+			block := test.BlockPairBuilder().Build()
+			driver.commitBlock(block)
+
+			output, err := driver.blockStorage.GetTransactionsBlockHeader(&services.GetTransactionsBlockHeaderInput{BlockHeight:1})
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output.TransactionsBlockHeader).To(Equal(block.TransactionsBlock.Header))
+			Expect(output.TransactionsBlockMetadata).To(Equal(block.TransactionsBlock.Metadata))
+			Expect(output.TransactionsBlockProof).To(Equal(block.TransactionsBlock.BlockProof))
+		})
+
+		It("blocks if requested block is in near future", func (done Done) {
+			driver := NewDriver()
+			driver.expectCommitStateDiff()
+
+			block := test.BlockPairBuilder().Build()
+			driver.commitBlock(block)
```

Comment:
why do you need the first commit?

Reply:
I have a use case that's most likely to happen: read from the end of the initialized chain.